casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala

```diff
@@ -223,8 +223,26 @@ class Initializing[F[_]
     def getMinBlockHeight: F[Long] = {
       val blockHashes = startBlock.justifications.map(_.latestBlockHash)
       for {
-        blocks    <- blockHashes.traverse(BlockStore[F].getUnsafe)
-        minHeight = if (blocks.isEmpty) 0L else blocks.map(ProtoUtil.blockNumber).min
+        lowestBlockHeights <- fs2.Stream
+                               .fromIterator(blockHashes.toIterator)
+                               .evalMap { BlockStore[F].getUnsafe(_) }
+                               .evalMap { b =>
+                                 val noSelfJustificationErr =
+                                   "No latest messages justifications downloaded when reconstructed Last Finalized State"
+                                 val selfJ =
+                                   ProtoUtil
+                                     .creatorJustification(b)
+                                     .map(_.latestBlockHash)
+                                     .get
+                                 BlockStore[F]
+                                   .get(selfJ)
+                                   .flatMap(_.liftTo(new Exception(noSelfJustificationErr)))
+                                   .map(_.body.state.blockNumber)
+                               }
+                               .compile
+                               .toList
+
+        minHeight = if (lowestBlockHeights.isEmpty) 0L else lowestBlockHeights.min
```

Comment:
I made this to complicated, this calculation of min height here is not really necessary, which I totally missed.

Here we have final result of requester with min height calculated in `lowerBound`.
https://github.com/rchain/rchain/blob/c06fb989d16568f578863d8e343013f5e704e420/casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala#L194-L197

So instead we can re-use it and not have same logic in two places.
```scala
// Receive the blocks and after populate the DAG
blockRequestAddDagStream = blockRequestStream.last.unNoneTerminate.evalMap { st =>
  populateDag(approvedBlock.candidate.block, st.lowerBound, st.heightMap)
}
```